src/python/pants/backend/docker/dockerfile_lockfile.txt

```diff
@@ -6,7 +6,7 @@
 # {
 #   "version": 2,
 #   "valid_for_interpreter_constraints": [
-#     "CPython>=3.6.1"
+#     "CPython>=3.7"
```

Comment:
Why this change?

Reply:
Ah, because `dockerfile_parser.py` is run with the PEX as its entrypoint, and thus was failing with Py36 because Py37 only features? 

I think that's reasonable for us to only allow running with Py37+ given that Pants itself requires Py37, so it should be discoverable in the repo.

Feels like a losing battle to instead try to enforce that `dockerfile_parser.py` is Py36+ rather than Py37+.

Reply:
Yes exactly. I tried it first, when it was just typing issues, but the dataclass..